docs/content/integrations/embedded-elt/dlt.mdx

```diff
@@ -228,9 +228,148 @@ And that's it! You should now have two assets that load data to corresponding Sn
+
+    return example_resource
+
+
+class CustomDagsterDltTranslator(DagsterDltTranslator):
+    def get_asset_key(self, resource: DagsterDltResource) -> AssetKey:
+        """Overrides asset key to be the dlt resource name."""
+        return AssetKey(f"{resource.name}")
+
+    def get_deps_asset_keys(self, resource: DagsterDltResource) -> Iterable[AssetKey]:
+        """Overrides upstream asset key to be a single source asset."""
+        return [AssetKey("common_upstream_dlt_dependency")]
+
+
+@dlt_assets(
+    name="example_dlt_assets",
+    dlt_source=example_dlt_source(),
+    dlt_pipeline=dlt.pipeline(
+        pipeline_name="example_pipeline_name",
+        dataset_name="example_dataset_name",
+        destination="snowflake",
+        progress="log",
+    ),
+    dlt_dagster_translator=CustomDagsterDltTranslator(),
+)
+def dlt_example_assets(context: AssetExecutionContext, dlt: DagsterDltResource):
+    yield from dlt.run(context=context)
+```
+
+In this example we customized the translator to change how the dlt assets name is defined, and we also hard-coded the asset dependency that is upstream of our assets to provide a fan-out model from a single dependency to our dlt assets.
+
+### Assigning metadata to upstream source assets
+
+A common question is how to define metadata on the source assets that area upstream of the dlt assets.
+
+This can be accomplished by defining a <PyObject object="SourceAsset" /> with a key that matches the one defined in the <PyObject module="dagster_embedded_elt.dlt" object="DagsterDltTranslator" method="get_deps_assets_keys" /> method.
+
+For example, let's say we have defined a set of dlt assets named `thinkific_assets`, we can iterate over those assets and derive a <PyObject object="SourceAsset" /> with attributes like `group_name`.
+
+```python file=/integrations/embedded_elt/dlt_source_assets.py
+import dlt
+from dagster_embedded_elt.dlt import DagsterDltResource, dlt_assets
+
+from dagster import AssetExecutionContext, SourceAsset
+
+
+@dlt.source
+def example_dlt_source():
+    def example_resource(): ...
+
+    return example_resource
+
+
+@dlt_assets(
+    dlt_source=example_dlt_source(),
+    dlt_pipeline=dlt.pipeline(
+        pipeline_name="example_pipeline_name",
+        dataset_name="example_dataset_name",
+        destination="snowflake",
+        progress="log",
+    ),
+)
+def example_dlt_assets(context: AssetExecutionContext, dlt: DagsterDltResource):
+    yield from dlt.run(context=context)
+
+
+thinkific_source_assets = [
+    SourceAsset(key, group_name="thinkific")
+    for key in example_dlt_assets.dependency_keys
+]
+```
+
+### Using partitions in your dlt assets
+
+While still an experimental feature, it is possible to use partitions within your dlt assets. However, it should be noted that this may result in concurrency related issues as state is managed by dlt. For this reason, it is recommended to set concurrency limits for your partitioned dlt assets.
```

Comment:
```suggestion
While still an experimental feature, it is possible to use partitions within your dlt assets. However, it should be noted that this may result in concurrency-related issues as state is managed by dlt. For this reason, it is recommended to set concurrency limits for your partitioned dlt assets.
```

Reply:
maybe link to concurrency in the docs?

Reply:
Good call, added in https://github.com/dagster-io/dagster/pull/22822/commits/28b24c49a445f8a0aa484b40c488826e1d85fbd3